lib/langchain/vectorsearch/epsilla.rb

```diff
@@ -0,0 +1,131 @@
+# frozen_string_literal: true
+
+require "securerandom"
+require "json"
+require "timeout"
+
+module Langchain::Vectorsearch
+  class Epsilla < Base
+    #
+    # Wrapper around Epsilla client library
+    #
+    # Gem requirements:
+    #     gem "epsilla-ruby", "~> 0.0.3"
+    #
+    # Usage:
+    #     epsilla = Langchain::Vectorsearch::Epsilla.new(protocol:, host:, db_name:, db_path:, index_name:, llm:, port:)
+    #
+    # Initialize Epsilla client
+    # @param protocol [String] The protocol to use, e.g. http or https
+    # @param host [String] The host to connect to
+    # @param db_name [String] The name of the database to use
+    # @param db_path [String] The path to the database to use
+    # @param index_name [String] The name of the Epsilla table to use
+    # @param llm [Object] The LLM client to use
+    # @param port [Integer] The port to connect to, default 8888
+    def initialize(protocol:, host:, db_name:, db_path:, index_name:, llm:, port: 8888)
```

Comment:
I would kind of prefer that the initializer just accepts `url:, llm:` params.

Would it then be possible to break the url into protocol, host, port, db_name, path, etc as needed?

Reply:
It's quite natural to use `url` to encode `protocol`, `host` and `port`, so I have made the change to simplify the params list.

However, Epsilla does not have a canonical way to represent `db_name` and `db_path` in a URL. Compared to other databases, it is a bit clunky. I thought it would be better to keep it explicit and avoid confusions.

What do you think?